draft-irtf-cfrg-opaque.md

```diff
@@ -692,163 +707,46 @@ Exceptions:
 Steps:
 1. auth_key = Expand(randomized_pwd, concat(envelope.nonce, "AuthKey"), Nh)
 2. export_key = Expand(randomized_pwd, concat(envelope.nonce, "ExportKey", Nh)
-3. (client_private_key, client_public_key) =
-    RecoverKeys(randomized_pwd, envelope.nonce, envelope.inner_env)
-4. cleartext_creds = CreateCleartextCredentials(server_public_key,
+3. seed = Expand(randomized_pwd, concat(envelope.nonce, "PrivateKey"), Nsk)
+4. client_private_key, client_public_key = DeriveAuthKeyPair(seed)
+5. cleartext_creds = CreateCleartextCredentials(server_public_key,
                       client_public_key, server_identity, client_identity)
-5. expected_tag = MAC(auth_key, concat(envelope.nonce, inner_env, cleartext_creds))
-6. If !ct_equal(envelope.auth_tag, expected_tag),
+6. expected_tag = MAC(auth_key, concat(envelope.nonce, inner_env, cleartext_creds))
+7. If !ct_equal(envelope.auth_tag, expected_tag),
      raise EnvelopeRecoveryError
-7. Output (client_private_key, export_key)
-~~~
-
-## Envelope Modes {#envelope-modes}
-
-The `EnvelopeMode` specifies the structure and encoding of the
-corresponding `InnerEnvelope`. This document specifies the values
-of the two aforementioned modes:
-
-~~~
-enum {
-  internal(1),
-  external(2),
-  (255)
-} EnvelopeMode;
+8. Output (client_private_key, export_key)
 ~~~
 
-Each `EnvelopeMode` defines its own `InnerEnvelope` structure and must implement
-the following interface:
+#### AKE compatibility
```

Comment:
Should this go in the AKE section? It seems kinda out of place here.

Reply:
Yes, and reference this bit from here 👍